### ISSUE TITLE
Unable to install avocado-framework-plugin-varianter-yaml-to-mux:102 cleanly

### ISSUE THREAD

**Issue by AYUSHJAIN951** [avocado-framework/avocado#5713]
While upgrading 101.0 ->102.0, avocado-framework-plugin-varianter-yaml-to-mux seems to be having version issues 

```
 git clone https://github.com/open-power-host-os/tests.git
 cd tests
 sed -i "s/work', ''/work', 'git+https:\/\/github.com\/avocado-framework\/avocado.git'/g" config/wrapper/env.conf
 head -15 config/wrapper/env.conf
 ./avocado-setup.py --bootstrap --install-deps
```

Ideally it should be installing avocado and avocado-framework-plugin-varianter-yaml-to-mux with 102.0 release

But currently it seems pip is not able to fetch and install  avocado-framework-plugin-varianter-yaml-to-mux:102.0 and instead installing 101.0 .
```
root@AMDtesthost_Ubuntu:~/tests# avocado -v
Traceback (most recent call last):
  File "/usr/local/bin/avocado", line 5, in <module>
    from avocado.core.main import main
  File "/usr/local/lib/python3.10/dist-packages/avocado/__init__.py", line 35, in <module>
    initialize_plugins()
  File "/usr/local/lib/python3.10/dist-packages/avocado/core/__init__.py", line 290, in initialize_plugins
    InitDispatcher().map_method("initialize")
  File "/usr/local/lib/python3.10/dist-packages/avocado/core/dispatcher.py", line 207, in __init__
    super().__init__("avocado.plugins.init")
  File "/usr/local/lib/python3.10/dist-packages/avocado/core/enabled_extension_manager.py", line 25, in __init__
    super().__init__(namespace, invoke_kwds)
  File "/usr/local/lib/python3.10/dist-packages/avocado/core/extension_manager.py", line 94, in __init__
    plugin = ep.load()
  File "/usr/local/lib/python3.10/dist-packages/pkg_resources/__init__.py", line 2517, in load
    self.require(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/pkg_resources/__init__.py", line 2540, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
  File "/usr/local/lib/python3.10/dist-packages/pkg_resources/__init__.py", line 829, in resolve
    dist = self._resolve_dist(
  File "/usr/local/lib/python3.10/dist-packages/pkg_resources/__init__.py", line 875, in _resolve_dist
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.VersionConflict: (avocado-framework 102.0 (/usr/local/lib/python3.10/dist-packages), Requirement.parse('avocado-framework==101.0'))
```

```
2023-06-27 06:27:13 DEBUG   : [helper.py:49] Running pip3 install -U avocado-framework-plugin-varianter-yaml-to-mux
2023-06-27 06:27:14 DEBUG   : [helper.py:55] Collecting avocado-framework-plugin-varianter-yaml-to-mux
  Downloading avocado_framework_plugin_varianter_yaml_to_mux-101.0-py3-none-any.whl (10 kB)
Collecting avocado-framework==101.0 (from avocado-framework-plugin-varianter-yaml-to-mux)
  Downloading avocado_framework-101.0-py3-none-any.whl (430 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 430.7/430.7 kB 3.0 MB/s eta 0:00:00
Requirement already satisfied: PyYAML>=4.2b2 in /usr/lib/python3/dist-packages (from avocado-framework-plugin-varianter-yaml-to-mux) (5.4.1)
Requirement already satisfied: setuptools in /usr/local/lib/python3.10/dist-packages (from avocado-framework==101.0->avocado-framework-plugin-varianter-yaml-to-mux) (68.0.0)
Installing collected packages: avocado-framework, avocado-framework-plugin-varianter-yaml-to-mux
  Attempting uninstall: avocado-framework
    Found existing installation: avocado-framework 102.0
    Uninstalling avocado-framework-102.0:
      Successfully uninstalled avocado-framework-102.0
Successfully installed avocado-framework-101.0 avocado-framework-plugin-varianter-yaml-to-mux-101.0
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
2023-06-27 06:27:14 DEBUG   : [helper.py:47] Installing python package avocado-framework-plugin-result-html using pip
2023-06-27 06:27:14 DEBUG   : [helper.py:49] Running pip3 install -U avocado-framework-plugin-result-html
2023-06-27 06:27:15 DEBUG   : [helper.py:55] Collecting avocado-framework-plugin-result-html
  Downloading avocado_framework_plugin_result_html-102.0-py3-none-any.whl (96 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 96.9/96.9 kB 1.4 MB/s eta 0:00:00
Collecting avocado-framework==102.0 (from avocado-framework-plugin-result-html)
  Using cached avocado_framework-102.0-py3-none-any.whl (439 kB)
Requirement already satisfied: jinja2 in /usr/lib/python3/dist-packages (from avocado-framework-plugin-result-html) (3.0.3)
Requirement already satisfied: setuptools in /usr/local/lib/python3.10/dist-packages (from avocado-framework==102.0->avocado-framework-plugin-result-html) (68.0.0)
Installing collected packages: avocado-framework, avocado-framework-plugin-result-html
  Attempting uninstall: avocado-framework
    Found existing installation: avocado-framework 101.0
    Uninstalling avocado-framework-101.0:
      Successfully uninstalled avocado-framework-101.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
avocado-framework-plugin-varianter-yaml-to-mux 101.0 requires avocado-framework==101.0, but you have avocado-framework 102.0 which is incompatible.
Successfully installed avocado-framework-102.0 avocado-framework-plugin-result-html-102.0
```

```
root@AMDtesthost_Ubuntu:~/tests# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04 LTS
Release:        22.04
Codename:       jammy

Installation Method : pip
```

**Comment by AYUSHJAIN951** [comment]
https://pypi.org/project/avocado-framework-plugin-varianter-yaml-to-mux/#history
I am unable to see 102.0 release of the following plugin
@clebergnu Can you please look at this issue.

**Comment by clebergnu** [comment]
> https://pypi.org/project/avocado-framework-plugin-varianter-yaml-to-mux/#history I am unable to see 102.0 release of the following plugin @clebergnu Can you please look at this issue.

Hi @AYUSHJAIN951 ,

I can confirm the issue you are reporting is valid.  The cause of the problem is that we moved the documentation to the plugin (in hope of giving them a better description).  But in the specific case of the yaml-to-mux plugin, there's directives that are Sphinx-only, and not pure ResTructuredText (rst).

We'll add checks so that it doesn't happen anymore, probably something like:

```
$ twine check avocado_framework_plugin_varianter_yaml_to_mux-102.0-py3-none-any.whl 
Checking avocado_framework_plugin_varianter_yaml_to_mux-102.0-py3-none-any.whl: FAILED
ERROR    `long_description` has syntax errors in markup and would not be rendered on PyPI.                                                                      
         line 6: Error: Unknown interpreted text role "mod".
```

I'll check if there's anything we can do to mitigate this to version 102.0 without requiring a new release.

**Comment by clebergnu** [comment]
Hi @AYUSHJAIN951 ,

I published a version of the plugins for 102.0 with just the long description modified.  You can find it [here](https://pypi.org/project/avocado-framework-plugin-varianter-yaml-to-mux/102.0/).  The installation should now proceed as expected.

I won't close this issue for now, because like I said before, we have to make sure this doesn't happen again.

Thanks!

**Comment by AYUSHJAIN951** [comment]
> Hi @AYUSHJAIN951 ,
> 
> I published a version of the plugins for 102.0 with just the long description modified. You can find it [here](https://pypi.org/project/avocado-framework-plugin-varianter-yaml-to-mux/102.0/). The installation should now proceed as expected.
> 

Thanks @clebergnu for the quick fix.

> I won't close this issue for now, because like I said before, we have to make sure this doesn't happen again.
> 

Sure let's keep the issue open for now until its having a proper fix.

> Thanks!

Thanks!!
